lib/honeybadger/instrumentation.rb

```diff
@@ -0,0 +1,113 @@
+module Honeybadger
+  # +Honeybadger::Instrumentation+ defines the API for collecting metric data from anywhere
+  # in an application. These class methods may be used directly, or you can include the
+  # Honeybadger::InstrumentationHelper for a convient wrapper within a class.
+  #
+  # @example
+  #
+  # class TicketsController < ApplicationController
+  #   def create
+  #     Honeybadger::Instrumentation.time('create.ticket', ->{
```

Comment:
Just curious, is there a reason why not to add these methods directly to `Honeybadger::Agent` and delegate them in [the `singleton`](https://github.com/honeybadger-io/honeybadger-ruby/blob/master/lib/honeybadger/singleton.rb) so that you can call `Honeybadger.time`, `Honeybadger.guage`, etc., similar to `Honeybadger.notify` and `Honeybadger.event`?

(I can't remember if we discussed this elsewhere, so apologies if I'm reopening the discussion. 😀 )

Reply:
Oh that's a great idea. I think that makes sense to me. I'm not sure if it was discussed or not, but I like it.